### ISSUE TITLE
Add ability to "auto-size" columns in Data Browser

### ISSUE THREAD

**Issue by NSExceptional** [sqlitebrowser/sqlitebrowser#2351]
There is currently no way to auto-size the columns when browsing table data as far as I can tell. This would be a nice QoL improvement.

**Comment by justinclift** [comment]
Yeah, agreed.  I keep trying to double click the border between cells, thinking it'll automatically resize that cell.  But it doesn't work.

Pretty sure this has been suggested before too, but I have no idea what became of the suggestion.  It'd be really useful. :smile:

**Comment by mgrojo** [comment]
The [columns can be resized](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Browse-Data#resizing-columns) to fit contents if the separator betweeen columns in the header is double-clicked. Moreover, in v3.12 the columns are sized by default to fit contents when the table is opened, until you save your preferences in a project file.

What could be implemented is a way to resize all the columns to fit their contents. Is that what you are asking for?

**Comment by jbhanks** [comment]
I too would love to be able to have the columns sized to the data rather than the column name as an option.
